beacon-chain/core/electra/attestation.go

```diff
@@ -1,7 +1,94 @@
 package electra
 
-import "github.com/prysmaticlabs/prysm/v5/beacon-chain/core/altair"
+import (
+	"context"
+	"fmt"
+
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/altair"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/time"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	customtypes "github.com/prysmaticlabs/prysm/v5/beacon-chain/state/state-native/custom-types"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1/attestation"
+)
 
 var (
 	ProcessAttestationsNoVerifySignature = altair.ProcessAttestationsNoVerifySignature
 )
+
+// GetProposerRewardNumerator returns the numerator of the proposer reward for an attestation.
+func GetProposerRewardNumerator(
+	ctx context.Context,
+	st state.ReadOnlyBeaconState,
+	att ethpb.Att,
+	totalBalance uint64,
+) (uint64, error) {
+	data := att.GetData()
+
+	delay, err := st.Slot().SafeSubSlot(data.Slot)
+	if err != nil {
+		return 0, fmt.Errorf("attestation slot %d exceeds state slot %d", data.Slot, st.Slot())
+	}
+
+	flags, err := altair.AttestationParticipationFlagIndices(st, data, delay)
+	if err != nil {
+		return 0, err
+	}
+
+	committees, err := helpers.AttestationCommitteesFromState(ctx, st, att)
```

Comment:
A nit, not sure if it's worth changing but this will fetch committees on a loop instead of getting the committees once. The loop is not large in Electra though